### DISCUSSION TITLE
Search in Google via Editor context menu

### DISCUSSION THREAD

**Discussion by AlexJuca** [zed-industries/zed#22050] [Show and tell]
Hey, it would be nice to have a Search with Google option in the context menu, similar to what exists in IntelliJ IDEA and CLion.

This is how it would look like:

<img width="799" alt="Screenshot 2024-12-15 at 21 09 59" src="https://github.com/user-attachments/assets/16a0f826-64a2-4dc4-8d7f-da94da7de4d2" />

--

I have created a PR to add this to Zed here:

https://github.com/zed-industries/zed/pull/22047?new_mergebox=true

**Comment by andelink** [comment]
You can achieve a decent enough alternative as a zed task. I use this:

```json
  {
    "label": "kagi: ${ZED_SELECTED_TEXT}",
    "command": "open",
    "args": ["https://kagi.com/search?q=${ZED_SELECTED_TEXT}"],
    "reveal": "never",
    "hide": "on_success",
    "show_summary": false,
    "show_command": false,
    "use_new_terminal": false,
    "shell": {
      "program": "sh"
    }
  }
```

You can replace kagi with google and it works fine.
